tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,217 @@
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/base/function_parser.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/base/convergence_table.h>
+#include "parameters/all_parameters.h"
+#include "parameters/parameters.h"
+#include "numerical_flux/numerical_flux_factory.hpp"
+#include "physics/physics_factory.h"
+#include "dg/dg.h"
+#include "dg/dg_factory.hpp"
+#include "ode_solver/ode_solver_factory.h"
+
+
+int main (int argc, char * argv[])
+{
+    dealii::Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+    const int dim = PHILIP_DIM;
+
+#if PHILIP_DIM==1
+    using Triangulation = dealii::Triangulation<PHILIP_DIM>;
+#else
+    using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+#endif
+
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+#if PHILIP_DIM!=1
+            MPI_COMM_WORLD,
+#endif
+
+            //Advects a sine wave (1D) with peridic BCs and compares to anlytical solution
+            //refines spatial discretization 5 times and writes a convergence table to file
+            //Can be used to verify spatial order of the explicit ODE solver
+            //Note: 2D is implemented but is not included in CMakeLists.txt (untested)
+
+
+            typename dealii::Triangulation<dim>::MeshSmoothing(
+                dealii::Triangulation<dim>::smoothing_on_refinement |
+                dealii::Triangulation<dim>::smoothing_on_coarsening));
+
+    int testfail = 0;
+
+    double left = 0.0;
+    double right = 2.0;
+    const bool colorize = true;
+    int n_refinements = 5;
+
+    //generating grid
+    dealii::GridGenerator::hyper_cube(*grid, left, right, colorize);
+    //setting periodic BC 
+    std::vector<dealii::GridTools::PeriodicFacePair<typename Triangulation::cell_iterator> > matched_pairs;
+    if (dim == 2){
```

Comment:
Since this test is only for dim = 1 or 2, it's better to give an error message if dim !=1,2.

Reply:
Thanks, I'll add an error check for `dim`.